transaction/jta/src/main/java/io/helidon/transaction/jta/JtaTxSupport.java

```diff
@@ -0,0 +1,575 @@
+/*
+ * Copyright (c) 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.transaction.jta;
+
+import java.lang.System.Logger.Level;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import io.helidon.common.Weight;
+import io.helidon.common.Weighted;
+import io.helidon.service.registry.Service;
+import io.helidon.transaction.Tx;
+import io.helidon.transaction.TxException;
+import io.helidon.transaction.spi.TxLifeCycle;
+import io.helidon.transaction.spi.TxSupport;
+
+import jakarta.transaction.HeuristicMixedException;
+import jakarta.transaction.HeuristicRollbackException;
+import jakarta.transaction.InvalidTransactionException;
+import jakarta.transaction.NotSupportedException;
+import jakarta.transaction.RollbackException;
+import jakarta.transaction.Status;
+import jakarta.transaction.SystemException;
+import jakarta.transaction.Transaction;
+import jakarta.transaction.TransactionManager;
+
+/**
+ * JTA transaction handling support.
+ * This is the default TxSupport service linking transaction API with JTA.
+ */
+@Service.Singleton
+@Weight(Weighted.DEFAULT_WEIGHT)
+class JtaTxSupport implements TxSupport {
+
+    private static final System.Logger LOGGER = System.getLogger(JtaTxSupport.class.getName());
+
+    private final JtaProvider provider;
+    private final List<TxLifeCycle> txListeners;
+
+    @Service.Inject
+    JtaTxSupport(JtaProvider provider, List<TxLifeCycle> txListeners) {
+        this.provider = provider;
+        this.txListeners = txListeners;
+    }
+
+    @Override
+    public String type() {
+        return "jta";
+    }
+
+    @Override
+    public <T> T transaction(Tx.Type type, Callable<T> task) {
+        start();
+        try {
+            return switch (type) {
+                case MANDATORY -> txMandatory(task);
+                case NEW -> txNew(task);
+                case NEVER -> txNever(task);
+                case REQUIRED -> txRequired(task);
+                case SUPPORTED -> txSupported(task);
+                case UNSUPPORTED -> txUnsupported(task);
+            };
+        // end() must always happen after start
+        } finally {
+            end();
+        }
+    }
+
+/* Removed: need to decide whether this is needed at all
```

Comment:
We don't normally include commented-out code

Reply:
Yes, it was deleted.